docs/DeveloperGuide.md

```diff
@@ -330,30 +330,308 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `ContactCS` and the **Actor** is the `CS freshman`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 Add a contact**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to add a new contact by inputting the relevant contact details (name, email, phone number, module code, etc.)
+2. ContactCS adds the contact under the specified module
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. Necessary input such as module code, name, or email is missing.
 
-  Use case ends.
+  * 1a1. ContactCS requests the user to provide the required information
+  * 1a2. User enters new data
+  * Steps 1a1 - 1a2 are repeated until the data entered are correct
+        
+    Use case resumes from step 2.
 
-* 3a. The given index is invalid.
+* 1b. The given format is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 1b1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 1b2. User enters new input
+    * Steps 1b1 - 1b2 are repeated until the format for the entered input is correct
 
-      Use case resumes at step 2.
+      Use case resumes from step 2.
+
+* 1c. The given contact is a duplicate.
+
+    * 1c1. ContactCS shows an error message telling the user that the contact already exists in ContactCS
+
+      Use case ends.
+
+<a name="uc02"></a>
+
+**Use case: UC02 Search contact**
+
+**MSS**
+
+1.  User requests to search for a contact by inputting exactly one of them:
+     * The name of the contact,
+     * The module code (optionally including the role), or
+     * The category of the issue
+2. ContactCS shows a list of matching contacts
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. Necessary input such as module code, or category of the issue is missing.
+
+    * 1a1. ContactCS requests the user to provide the required information
+    * 1a2. User enters new input
+    * Steps 1a1 - 1a2 are repeated until all required fields in the input are complete
+
+      Use case resumes from step 2.
+
+* 1b. The given format is invalid.
+
+    * 1b1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 1b2. User enters new input
+    * Steps 1b1 - 1b2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+* 1c. No matching contacts found.
+
+    Use case ends.
```

Comment:
Would it be better to display a message here as well? Just like empty list in other use cases